runtime/src/iree/hal/drivers/cuda/dynamic_symbols.c

```diff
@@ -127,6 +128,38 @@ iree_status_t iree_hal_cuda_nccl_dynamic_symbols_initialize(
   if (iree_status_is_ok(status)) {
     status = iree_hal_cuda_nccl_dynamic_symbols_resolve_all(out_syms);
   }
+
+  // Check the NCCL version compatibility
+  int nccl_version;
+  if (iree_status_is_ok(status)) {
+    status = NCCL_RESULT_TO_STATUS(out_syms, ncclGetVersion(&nccl_version));
+  }
+
+  if (iree_status_is_ok(status)) {
+    int major, minor;
+
+    if (nccl_version < 20000) {
```

Comment:
unfortunate they don't include these in their own header; you could simplify this to `if (nccl_version < NCCL_VERSION(NCCL_MAJOR, NCCL_MINOR, 0)) return ...` to avoid needing to do the math yourself, though you may still want to do it for better error messages - I'm not confident they won't change the scheme again in the future, though :)

Reply:
Yeah, I have to get major and minor for better user message, unfortunately.

Reply:
You could add a check here that you've computed these correctly by checking
`NCCL_VERSION(major, minor, patch) == ncl_version` and adding to the error message if it isn't